### PR TITLE
Enable keycloak authentication to fire weather page WPS #193

### PR DESCRIPTION
**Pull request by ghost**
WPS #193

### CODE REVIEW

**Review by DerekRoberts — DISMISSED** [review]

File: src/components/PageHeader.tsx [hunk 1]
```diff
@@ -59,14 +60,24 @@ export const PageHeader = ({ title }: Props) => {
...
-          className={classes.contact}
-          href={process.env.REACT_APP_FIDER_LINK}
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Contact
-        </a>
+        <div>
+          <a
+            className={classes.contact}
```
Comment (Sybrand) [inline src/components/PageHeader.tsx:65]: We haven't decided that we're going to have a link to the fire weather calculator, and it's not part of the acceptance criteria of this story.

File: src/utils/keycloak.example.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+    "initOptions": {
+    "url": "https://sso.pathfinder.gov.bc.ca/auth",
```
Comment (Sybrand) [inline src/utils/keycloak.example.json:3]: This has to be different for dev/test/prod - so it can't be in a hard coded json file. Please move to environment variables.

File: package.json [hunk 3]
```diff
@@ -17,15 +17,19 @@
...
     "@reduxjs/toolkit": "^1.2.5",
+    "@testing-library/react": "^9.5.0",
     "@types/node": "^12.12.26",
     "@types/react": "^16.9.19",
     "@types/react-dom": "^16.9.5",
     "@types/react-redux": "^7.1.7",
+    "@types/react-router-dom": "^5.1.3",
     "@types/webpack-env": "^1.15.1",
     "axios": "^0.19.2",
+    "keycloak-js": "^9.0.2",
```
Comment (Sybrand) [inline package.json:28]: Does this pull keycloak.js from https://sso.pathfinder.gov.bc.ca/auth/js/keycloak.js ? This is important, as the version of the client has to match that of the server.

**Review by Sybrand — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (Nkuruba) [reply package.json:28]: @Sybrand no but react has an official keycloak library to manage calls to keycloak server.

File: src/components/PageHeader.tsx (hunk 1 above)
Comment (Nkuruba) [reply src/components/PageHeader.tsx:65]: @Sybrand the link is for temporary purpose as it would be removed as part of next coming PRs. The link was only introduced to showcase the navigation and login functionality.

File: src/components/PageHeader.tsx [hunk 4]
```diff
@@ -59,14 +60,24 @@ export const PageHeader = ({ title }: Props) => {
...
+            href="/fireWeatherCalculator"
+            target="_blank"
+            rel="noopener noreferrer"
+          >
+            FireWeather
+          </a>
```
Comment (Kyubinhan) [inline src/components/PageHeader.tsx:71]: Do you think we need this @Sybrand ? Cause my understanding was each product(pages) are gonna be independent

File: src/utils/keycloak.example.json (hunk 2 above)
Comment (Nkuruba) [reply src/utils/keycloak.example.json:3]: @Sybrand this is an example file and it was added only to run the app locally.

**Review by Kyubinhan — CHANGES_REQUESTED** [review]

File: src/utils/keycloak.example.json (hunk 2 above)
Comment (Kyubinhan) [reply src/utils/keycloak.example.json:3]: Can you then change it to `https://sso-dev.pathfinder.gov.bc.ca/auth`

File: openshift/templates/deploy.dc.yaml [hunk 5]
```diff
@@ -40,12 +40,26 @@ parameters:
   - name: MEMORY_LIMIT
     description: Memory upper limit
     value: 2Gi
+  - name: KEYCLOAK_URL_AUTH
+    value: https://sso.pathfinder.gov.bc.ca/auth
+  - name: KEYCLOAK_REALM
+    value: 8wl6x4cp
+  - name: KEYCLOAK_CLIENT
+    value: security-admin-console
```
[the remark this answers is not in the document]
Comment (Nkuruba) [reply openshift/templates/deploy.dc.yaml:48]: @Kyubinhan yes but it was not part of acceptance criteria so I managed to use the current one. We can create new one as part of a different ticket.

File: src/features/fwCalculator/FWCalculatorPage.tsx [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+          setAuthenciation(authenticated)
+        }
+        setTimeout(() => {
+          keycloak.updateToken(60)
+        }, 60000)
+      })
+  }, [])
+
+  if (isAuthenciated) return <header>Hello World!!!</header>
+  else return <div />
```
Comment (Kyubinhan) [inline src/features/fwCalculator/FWCalculatorPage.tsx:24]: We need to store `isAuthenciated` in Redux not in local state

File: package.json [hunk 7]
```diff
@@ -17,15 +17,19 @@
     "@material-ui/icons": "^4.9.1",
     "@material-ui/lab": "^4.0.0-alpha.43",
     "@reduxjs/toolkit": "^1.2.5",
+    "@testing-library/react": "^9.5.0",
```
[the remark this answers is not in the document]
Comment (Nkuruba) [reply package.json:20]: @Kyubinhan the app was not starting as it was missing this library and I had to add it to resolve the issue.

File: src/utils/keycloak.example.json (hunk 2 above)
Comment (Nkuruba) [reply src/utils/keycloak.example.json:3]: ok

File: openshift/templates/deploy.dc.yaml [hunk 8]
```diff
@@ -40,12 +40,26 @@ parameters:
   - name: MEMORY_LIMIT
     description: Memory upper limit
     value: 2Gi
+  - name: KEYCLOAK_URL_AUTH
+    value: https://sso.pathfinder.gov.bc.ca/auth
```
[the remark this answers is not in the document]
Comment (Nkuruba) [reply openshift/templates/deploy.dc.yaml:44]: @Kyubinhan this is a pipeline design related issue as we are not following different URL's for API as well. We are using prod API URL by default in the template and I had to follow same approach. We can discuss on this and implement it as part of a new ticket to separate out config for different environment.

File: openshift/templates/deploy.dc.yaml [hunk 9]
```diff
@@ -40,12 +40,26 @@ parameters:
   - name: MEMORY_LIMIT
     description: Memory upper limit
     value: 2Gi
+  - name: KEYCLOAK_URL_AUTH
```
[the remark this answers is not in the document]
Comment (Nkuruba) [reply openshift/templates/deploy.dc.yaml:43]: @Kyubinhan my idea of using this variable was to convey that it is keycloak url for authentication so I want to stick with  KEYCLOAK_URL_AUTH.

File: src/app/App.tsx [hunk 10]
```diff
@@ -40,7 +42,18 @@ const App = () => {
...
-        {showModal ? <FWICalculatorDisclaimerModal /> : <FWICalculatorPage />}
+        <Router>
+          <Route path="/" exact>
+            {showModal ? (
+              <FWICalculatorDisclaimerModal />
+            ) : (
+              <FWICalculatorPage />
+            )}
+          </Route>
+          <Route path="/fireWeatherCalculator" exact>
```
[the remark this answers is not in the document]
Comment (Nkuruba) [reply src/app/App.tsx:53]: ok

File: src/app/App.tsx [hunk 11]
```diff
@@ -40,7 +42,18 @@ const App = () => {
     <React.StrictMode>
       <CssBaseline />
       <ThemeProvider theme={theme}>
-        {showModal ? <FWICalculatorDisclaimerModal /> : <FWICalculatorPage />}
+        <Router>
+          <Route path="/" exact>
```
[the remark this answers is not in the document]
Comment (Nkuruba) [reply src/app/App.tsx:46]: @Kyubinhan why do you need to redirect users from calculator page?

File: openshift/templates/deploy.dc.yaml (hunk 8 above)
[the remark this answers is not in the document]
Comment (Sybrand) [reply openshift/templates/deploy.dc.yaml:44]: Agreed - same as per json file.

File: openshift/templates/deploy.dc.yaml (hunk 9 above)
[the remark this answers is not in the document]
Comment (Sybrand) [reply openshift/templates/deploy.dc.yaml:43]: Agreed.

File: src/components/PageHeader.tsx (hunk 4 above)
Comment (Sybrand) [reply src/components/PageHeader.tsx:71]: We explicitly agreed that they were going to be independent, so agreed - I don't know where this came from.

Comment (Nkuruba) [reply src/components/PageHeader.tsx:71]: @Sybrand @Kyubinhan yes fire weather is a independent component. I just added this link to as as a temporary solution. This will be removed when actual routing would be added to the application. There is already another ticket for this purpose wps #196.

File: package.json (hunk 3 above)
Comment (Sybrand) [reply package.json:28]: If the keycloak client does not match the keycloak server version - there will be issues. This was clearly specified in the acceptance criteria, and clearly stated as best practice on the official documentation.

For more detail, please refer to the official documentation: https://www.keycloak.org/docs/latest/securing_apps/#_javascript_adapter

"A best practice is to load the JavaScript adapter directly from Keycloak Server as it will automatically be updated when you upgrade the server. If you copy the adapter to your web application instead, make sure you upgrade the adapter only after you have upgraded the server."

File: src/utils/keycloak.example.json (hunk 2 above)
Comment (Sybrand) [reply src/utils/keycloak.example.json:3]: My bad - sorry @Nkuruba

File: src/components/PageHeader.tsx (hunk 1 above)
Comment (Sybrand) [reply src/components/PageHeader.tsx:65]: We can demonstrate this by linking directly to a url know to us - please remove it from this PR as it ads unwanted functionality.

File: openshift/templates/deploy.dc.yaml (hunk 8 above)
[the remark this answers is not in the document]
Comment (Sybrand) [reply openshift/templates/deploy.dc.yaml:44]: i.m.h.o - this has to be solved as part of this PR.

File: openshift/templates/deploy.dc.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (Sybrand) [reply openshift/templates/deploy.dc.yaml:48]: @Kyubinhan do you mean a new realm entirely?

File: package.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (Sybrand) [reply package.json:20]: Please resolve this issue without including developer dependancies.

File: openshift/templates/deploy.dc.yaml (hunk 9 above)
[the remark this answers is not in the document]
Comment (Nkuruba) [reply openshift/templates/deploy.dc.yaml:43]: ok

File: openshift/templates/deploy.dc.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (Kyubinhan) [reply openshift/templates/deploy.dc.yaml:48]: @Sybrand No. Clients are part sso realms.

File: src/components/PageHeader.tsx (hunk 1 above)
Comment (Nkuruba) [reply src/components/PageHeader.tsx:65]: ok will remove it.

File: openshift/templates/deploy.dc.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (Sybrand) [reply openshift/templates/deploy.dc.yaml:48]: @Kyubinhan phew - ok - yes - agreed, we need a different client. And no, @Nkuruba not as part of a separate PR. This is not the security-admin-console, this is something else.

File: src/features/fwCalculator/FWCalculatorPage.tsx (hunk 6 above)
Comment (Nkuruba) [reply src/features/fwCalculator/FWCalculatorPage.tsx]: @Kyubinhan the authentication state is not being stored in redux store

File: src/components/PageHeader.tsx (hunk 4 above)
Comment (Nkuruba) [reply src/components/PageHeader.tsx]: @Sybrand @Kyubinhan I removed the link.

File: package.json (hunk 3 above)
Comment (Nkuruba) [reply package.json]: @Sybrand added code to read from keycloak.js on the fly.

File: openshift/templates/deploy.dc.yaml (hunk 8 above)
[the remark this answers is not in the document]
Comment (Nkuruba) [reply openshift/templates/deploy.dc.yaml]: @Sybrand @Kyubinhan code added to handle this.

File: openshift/templates/deploy.dc.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (Nkuruba) [reply openshift/templates/deploy.dc.yaml]: @Sybrand @Kyubinhan created wps-web client in all envs.

File: openshift/templates/deploy.dc.yaml [hunk 12]
```diff
@@ -40,12 +40,27 @@ parameters:
   - name: MEMORY_LIMIT
     description: Memory upper limit
     value: 2Gi
+  - name: KEYCLOAK_AUTH_URL
+    description: Keycloak url for authentication
+    required: true
```
Comment (DerekRoberts) [inline openshift/templates/deploy.dc.yaml]: Please replace `required: true` with `value: https://DEV_ADDRESS`.

File: openshift/templates/deploy.dc.yaml [hunk 13]
```diff
@@ -72,6 +87,10 @@ objects:
           containers:
             - image: ${DOCKER_REGISTRY}/${PROJECT_TOOLS}/${NAME}:${SUFFIX}
               imagePullPolicy: Always
+              volumeMounts:
+                - name: ${NAME}-${SUFFIX}
+                  mountPath: /app/src/utils/keycloak.json
```
Comment (Kyubinhan) [inline openshift/templates/deploy.dc.yaml:92]: Can you move 3 variables stored in the `keycloak.json` into `.env` file and use them just like how we use other env variables. I don't see any reason that they are being treated differently.

File: openshift/scripts/common/envars [hunk 14]
```diff
@@ -8,3 +8,5 @@ PROJ_PROD="${PROJ_PROD:-auzhsi-prod}"
 TAG_PROD="${TAG_PROD:-prod}"
 PATH_BC="${PATH_BC:-$(dirname ${0})/../templates/build.bc.yaml}"
 PATH_DC="${PATH_DC:-$(dirname ${0})/../templates/deploy.dc.yaml}"
+KEYCLOAK_PROD_URL="https://sso.pathfinder.gov.bc.ca/auth"
+KEYCLOAK_DEV_URL="https://sso-dev.pathfinder.gov.bc.ca/auth"
```
Comment (Kyubinhan) [inline openshift/scripts/common/envars:12]: I would name them SSO_BASE_PROD_URL and SSO_BASE_DEV_URL and btw I think you can just set the dev url here and use `https://sso.pathfinder.gov.bc.ca/auth` as a default in the template

**Review by DerekRoberts — DISMISSED** [review]
Anything that touches deployment and the pipeline looks good!  :)

File: src/features/fireWeather/FireWeatherPage.jsx [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+          }
+          setTimeout(() => {
+            keycloak.updateToken(60)
+          }, 6000)
+        })
+    }
+  }, [dispatch])
+
+  if (isAuthenticated) return <header>Hello World!!!</header>
+  else return <div />
```
Comment (Kyubinhan) [inline src/features/fireWeather/FireWeatherPage.jsx:45]: Can you render a message showing what state users are in? Rendering an empty <div> tag is not user-friendly.

File: src/features/fireWeather/slices/authenticationSlice.ts [hunk 16]
```diff
@@ -0,0 +1,58 @@
...
+export const createAuthentication = (
+  authenticated: boolean
+): AppThunk => async dispatch => {
+  try {
+    dispatch(getAuthenticationStart())
+    dispatch(getAuthenticationSuccess(authenticated))
+  } catch (err) {
+    dispatch(getAuthenticationFailed(err.toString()))
+  }
+}
```
Comment (Kyubinhan) [inline src/features/fireWeather/slices/authenticationSlice.ts:58]: I don't see any purpose of this action creator because there is no side effect involved. Side effect is actually happening when you run promise in `FireWeatherPage.jsx`

File: src/features/fireWeather/FireWeatherPage.jsx [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+          onLoad: 'login-required',
+          promiseType: 'native',
+          checkLoginIframe: false
+        })
+        .then(authenticated => {
+          if (!authenticated) {
+            dispatch(resetAuthentication())
+            window.location.reload()
+          } else {
+            dispatch(createAuthentication(authenticated))
```
Comment (Kyubinhan) [inline src/features/fireWeather/FireWeatherPage.jsx:35]: Dispatch getAuthenticationSuccess here. More explanation written in the slice file

File: src/features/fireWeather/FireWeatherPage.jsx [hunk 18]
```diff
@@ -0,0 +1,46 @@
...
+          }
+          setTimeout(() => {
+            keycloak.updateToken(60)
+          }, 6000)
+        })
```
Comment (Kyubinhan) [inline src/features/fireWeather/FireWeatherPage.jsx:40]: Use `.catch` here if you want to dispatch `getAuthenticationFailed` More explanation written in the slice file

**Review by Kyubinhan — CHANGES_REQUESTED** [review]

File: openshift/templates/deploy.dc.yaml (hunk 13 above)
Comment (Nkuruba) [reply openshift/templates/deploy.dc.yaml:92]: @Kyubinhan I dont want to introduce more environment variables. Also please go through https://www.keycloak.org/docs/latest/securing_apps/#_javascript_adapter as I am using same concept.

File: src/features/fireWeather/FireWeatherPage.jsx [hunk 19]
```diff
@@ -0,0 +1,46 @@
...
+  const { isAuthenticated } = useSelector(selectAuthenticationReducer)
+  const dispatch = useDispatch()
+
+  useEffect(() => {
+    const script = document.createElement('script')
+
+    script.src = `${initOptions.url}/js/keycloak.js`
+    script.async = true
+
+    const keyCloakScript = document.body.appendChild(script)
```
Comment (Kyubinhan) [inline src/features/fireWeather/FireWeatherPage.jsx:20]: You are applying this script every time useEffect is being run, so move this out from React related stuff

File: openshift/scripts/common/envars (hunk 14 above)
Comment (Nkuruba) [reply openshift/scripts/common/envars:12]: @Kyubinhan I want to stick with KEYCLOAK naming convention rather than SSO. Also I dont want app to point to prod by default.

File: src/features/fireWeather/FireWeatherPage.jsx (hunk 15 above)
Comment (Nkuruba) [reply src/features/fireWeather/FireWeatherPage.jsx:45]: @Kyubinhan I am not routing users to empty page with just <Div/>. This is just on way login flow.

File: openshift/templates/deploy.dc.yaml (hunk 13 above)
Comment (Kyubinhan) [reply openshift/templates/deploy.dc.yaml:92]: I don't see anywhere not to use environment variables in this docs. You can still initiate keycloak instance the way you want, but I am suggesting to reference env variables because I don't want to add a file coming from a configmap in build time later when we go production mode. You can mount it like this `/app/src/utils/keycloak.json` because we are running the app in dev mode

Comment (Nkuruba) [reply openshift/templates/deploy.dc.yaml:92]: @Kyubinhan the config map will be mounted during deployment time rather build time. Also I have already suggested that there may be more params coming in as part of Keycloak config and I dont want whole lot of env variables floating everywhere.

File: src/features/fireWeather/FireWeatherPage.jsx (hunk 15 above)
Comment (Kyubinhan) [reply src/features/fireWeather/FireWeatherPage.jsx:45]: I am not talking about routing. I am pointing out that in this if condition there is a change to render an empty div tag.

File: src/features/fireWeather/slices/authenticationSlice.ts (hunk 16 above)
Comment (Nkuruba) [reply src/features/fireWeather/slices/authenticationSlice.ts:58]: @Kyubinhan I placed it thinking we can use it if there were any side effects later so that no need to write this function entirely from scratch. Anyway I have removed it.

File: src/features/fireWeather/FireWeatherPage.jsx (hunk 17 above)
Comment (Nkuruba) [reply src/features/fireWeather/FireWeatherPage.jsx:35]: @Kyubinhan Updated it.

File: openshift/templates/deploy.dc.yaml (hunk 13 above)
Comment (Kyubinhan) [reply openshift/templates/deploy.dc.yaml:92]: Currently yes the config map is mounted during deployment, but your strategy won't work if we build the app into static files during the build time when we go to prod using caddy. Take a good look at Devhub's DC and BC because this is what we are aiming for. https://github.com/bcgov/devhub-app-web/blob/master/openshift/templates/dc.yaml

File: src/features/fireWeather/FireWeatherPage.jsx (hunk 18 above)
Comment (Nkuruba) [reply src/features/fireWeather/FireWeatherPage.jsx:40]: @Kyubinhan added catch.

File: src/features/fireWeather/FireWeatherPage.jsx (hunk 19 above)
Comment (Nkuruba) [reply src/features/fireWeather/FireWeatherPage.jsx]: @Kyubinhan done

File: public/index.html [hunk 20]
```diff
@@ -20,6 +20,8 @@
     <link rel="icon" href="%PUBLIC_URL%/images/bcid-favicon-16x16.png" sizes="16x16" type="image/png" />    
     <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons" />
 
+    <script src="%REACT_APP_KEYCLOAK_AUTH_URL%/js/keycloak.js"></script>
```
Comment (Kyubinhan) [inline public/index.html:23]: Nice!

File: .gitignore [hunk 21]
```diff
@@ -78,4 +78,7 @@ jspm_packages/
 .DS_Store
 
 # IDE - VSCode
-.vscode/*
\ No newline at end of file
+.vscode/*
+
+# keycloak json
+keycloak.json
```
Comment (Kyubinhan) [inline .gitignore:84]: Can you remove these lines?

File: README.md [hunk 22]
```diff
@@ -22,7 +22,7 @@ Installs all dependencies in the node_modules folder.
 
 ### Executing program
 
-In the project directory, create `.env` file at root using `.env.example` as a sample, then you can run:
+In the project directory, create `.env` file at root using `.env.example` as a sample and then you can run:
```
Comment (Kyubinhan) [inline README.md:25]: Can you revert this change?

File: src/features/fireWeather/FireWeatherPage.jsx [hunk 23]
```diff
@@ -0,0 +1,50 @@
...
+          dispatch(getAuthenticationSuccess(authenticated))
+        }
+        setTimeout(() => {
+          try {
+            keycloak.updateToken(60)
+          } catch (err) {
+            dispatch(getAuthenticationFailed(err))
+          }
+        }, 6000)
+      })
```
Comment (Kyubinhan) [inline src/features/fireWeather/FireWeatherPage.jsx:40]: What I was asking for was actually using Promise chaining.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises.
So right after then() block you can add .catch

**Review by Kyubinhan — CHANGES_REQUESTED** [review]
Few more changes please

File: openshift/scripts/common/envars [hunk 24]
```diff
@@ -8,3 +8,5 @@ PROJ_PROD="${PROJ_PROD:-auzhsi-prod}"
 TAG_PROD="${TAG_PROD:-prod}"
 PATH_BC="${PATH_BC:-$(dirname ${0})/../templates/build.bc.yaml}"
 PATH_DC="${PATH_DC:-$(dirname ${0})/../templates/deploy.dc.yaml}"
+KEYCLOAK_PROD_URL="https://sso.pathfinder.gov.bc.ca/auth"
```
Comment (Sybrand) [inline openshift/scripts/common/envars:11]: @Nkuruba @Kyubinhan we're being silly (re. our discussion about this yesterday) - and sorry if I missed the plot yesterday - but we don't need this at all. The DC should be pulling this from a configmap - so we don't have to specify the url here at all.

File: openshift/scripts/oc_deploy.sh [hunk 25]
```diff
@@ -21,13 +21,20 @@ source "$(dirname ${0})/common/common"
 #%   ${THIS_FILE} pr-0 apply
 #%
 
+# Use keycloak prod url when deploying to prod
```
Comment (Sybrand) [inline openshift/scripts/oc_deploy.sh:24]: @Nkuruba @Kyubinhan again - we don't need this here at all - we don't need to pass it to the template, the template can grab the values from the configmap.

File: openshift/scripts/oc_deploy.sh [hunk 26]
```diff
@@ -21,13 +21,20 @@ source "$(dirname ${0})/common/common"
...
+fi
+
 # Target project override for Dev or Prod deployments
 #
 PROJ_TARGET="${PROJ_TARGET:-${PROJ_DEV}}"
 
 # Process a template (mostly variable substition)
 #
-OC_PROCESS="oc -n ${PROJ_TOOLS} process -f ${PATH_DC} -p NAME=${NAME_APP} -p SUFFIX=${SUFFIX}"
+OC_PROCESS="oc -n ${PROJ_TOOLS} process -f ${PATH_DC} -p NAME=${NAME_APP} -p SUFFIX=${SUFFIX} -p KEYCLOAK_AUTH_URL=${KEYCLOAK_AUTH_URL}"
```
Comment (Sybrand) [inline openshift/scripts/oc_deploy.sh:37]: Please remove this - sorry for asking for it yesterday - I was totally wrong - this doesn't have to be passed to the template - we need a global configmap in our dev project that has all of these things specified.

File: src/features/fireWeather/FireWeatherPage.jsx [hunk 27]
```diff
@@ -0,0 +1,49 @@
...
+  resetAuthentication,
+  getAuthenticationFailed
+} from './slices/authenticationSlice'
+
+export const FireWeatherPage = () => {
+  const { isAuthenticated } = useSelector(selectAuthenticationReducer)
+  const dispatch = useDispatch()
+
+  useEffect(() => {
+    const keycloak = window.Keycloak({
```
Comment (Sybrand) [inline src/features/fireWeather/FireWeatherPage.jsx:15]: This is a good solution to start with - we'll need to revisit the moment we have more than one page needing logins, and I suspect we're not dealing correctly with tokens going stale etc. but that's a problem for another day.

File: public/index.html (hunk 20 above)
Comment (Sybrand) [inline public/index.html:23]: Fine for now, but we're going to have to solve async loading at some point - I'll write a ticket for it.

**Review by Sybrand — CHANGES_REQUESTED** [review]
Some comments - but only one change we need:
We can rely on config maps - we don't need to pollute things with dev/prod environment variables - sorry that I missed that yesterday when we walked through the code.

File: public/index.html (hunk 20 above)
Comment (Kyubinhan) [reply public/index.html:23]: By putting this script tag in `<head>`tag we don't have to worry about loading issue because this will be loaded first before React related scripts

Comment (Sybrand) [reply public/index.html:23]: It is going to slow down the overall loading of the page though. On second thought, it's not even worth a ticket - but we can get faster loading overall by loading it asynchronously and putting the wait in where we're actually needing to auth. BUT - REALLY - the file is going to be cached by the browser, so never mind - too much effort for speeding up the initial load.

File: openshift/scripts/oc_deploy.sh (hunk 25 above)
Comment (Nkuruba) [reply openshift/scripts/oc_deploy.sh]: @Sybrand now the template is grabbing values from global configmap.

**Comment by Nkuruba** [comment]
> Some comments - but only one change we need:
> We can rely on config maps - we don't need to pollute things with dev/prod environment variables - sorry that I missed that yesterday when we walked through the code.

@Sybrand now the deployment template uses values from global config map.

**Review by Sybrand — CHANGES_REQUESTED** [review]
Something seems to have gone wrong with deployment, I suspect you're going to have to close this PR and do a new one. (I think the failed production push did some cleanup that's now causing the build to fail?)

**Comment by Nkuruba** [comment]
> Something seems to have gone wrong with deployment, I suspect you're going to have to close this PR and do a new one. (I think the failed production push did some cleanup that's now causing the build to fail?)

@Sybrand could you please elaborate? where is the build failing?

File: openshift/templates/deploy.dc.yaml [hunk 28]
```diff
@@ -40,16 +36,9 @@ parameters:
   - name: MEMORY_LIMIT
     description: Memory upper limit
     value: 2Gi
+  - name: WPS_WEB_GLOBAL_PROPS
```
Comment (Kyubinhan) [inline openshift/templates/deploy.dc.yaml:39]: Can you rename this to `WPS_WEB_GLOBAL_CONFIG` or `WPS_WEB_GLOBAL_CONFIGMAP`

Comment (Nkuruba) [reply openshift/templates/deploy.dc.yaml]: done

**Review by Kyubinhan — DISMISSED** [review]

**Review by Sybrand — DISMISSED** [review]

**Review by DerekRoberts — APPROVED** [review]
I think this is ready to go!  Good job surviving us.  ;)
